### PR TITLE
Release UI: promote channel

### LINKED ISSUE #1119: Promote whole channel

### PR DESCRIPTION
**Pull request by bartaz**
Fixes #1119

- Adds possibility to promote whole channel at once (all architectures) - button with dropdown
- Removes buttons from single revisions (we will introduce checkboxes later)

### QA
- ./run or http://snapcraft.io-pr-1120.run.demo.haus/
- go to Release page of any snap
- click on arrow button next to channel to prepare for release of whole channel

screen shot 2018-09-20 at 18 23 30

### CODE REVIEW

**Comment by Lukewh** [comment]
I can't seem to click any of the items in the dropdown? I just get the regular text cursor. I've tried with multiple snaps.

**Comment by bartaz** [comment]
@Lukewh Thanks, good catch. Seems like vanilla contextual menu doesn't work on `<button>` element in Firefox. Will sort it out.

File: static/js/publisher/release/promoteButton.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+export default class PromoteButton extends Component {
+  componentDidMount() {
+    // use window instead of document, as React catches all events in document
+    window.addEventListener('click', this.closeAllDropdowns);
```
Comment (Lukewh) [inline static/js/publisher/release/promoteButton.js]: This seems to be a pattern we'll need down the line. Juju-gui implemented a wrapper component that allowed the click-outside to be added easily to any component. We should think about doing something similar.

File: static/sass/_snapcraft_release.scss [hunk 2]
```diff
@@ -59,4 +67,24 @@
...
     top: 10px;
   }
+
+  .p-channel-buttons {
+    display: inline-block;
+    margin-right: .5rem;
+    width: 2rem;
+  }
+
+
```
Comment (Lukewh) [inline static/sass/_snapcraft_release.scss]: empty line

File: static/js/publisher/release/promoteButton.js (hunk 1 above)
Comment (bartaz) [reply static/js/publisher/release/promoteButton.js]: I've seen some npm module for react that does more or less that, we can have a look, or see what juju implemented, or build our own component wrapper.

File: static/sass/_snapcraft_release.scss (hunk 2 above)
Comment (bartaz) [reply static/sass/_snapcraft_release.scss]: If only linter would catch things like that. I'll fix that with next release UI updates.

**Pull request merged**